sdcm/nemesis.py

```diff
@@ -2734,11 +2734,21 @@ def _disrupt_grow_shrink_cluster(self, rack=0):
                             end=end_time, duration=time_elapsed, **log_info).publish()
 
         self._set_current_disruption("ShrinkCluster")
-        self.log.info("Start shrink cluster on %s nodes", add_nodes_number)
+        # Check that number of nodes is enough for decommission:
+        cur_num_nodes_in_dc = len([n for n in self.cluster.nodes if n.dc_idx == self.target_node.dc_idx])
+        initial_db_size = self.tester.params.get("n_db_nodes")
+        if isinstance(initial_db_size, int):
+            decommission_nodes_number = min(cur_num_nodes_in_dc - initial_db_size, add_nodes_number)
+        else:
+            initial_db_size_in_dc = int(initial_db_size.split(" ")[self.target_node.dc_idx])
+            decommission_nodes_number = min(cur_num_nodes_in_dc - initial_db_size_in_dc, add_nodes_number)
+
+        self.log.info("Start shrink cluster on %s nodes", decommission_nodes_number)
```

Comment:
i would add here a check if `decommission_nodes_number` is bigger than zero, as otherwise, we can already skip (return with message "not enough resources to decommission nodes")

Reply:
Added